packages/ui/ant-design/src/ConnectionModal.tsx

```diff
@@ -0,0 +1,67 @@
+import { useSetConnection } from '@solana/wallet-adapter-react';
+import { Menu, Modal, ModalProps } from 'antd';
+import React, { FC, MouseEvent, useCallback, useMemo, useState } from 'react';
+import { useConnectionModal } from './useConnectionModal';
+import { ConnectionMenuItem } from './ConnectionMenuItem';
+import { Connection } from '@solana/web3.js';
+
+export interface ConnectionModalProps extends Omit<ModalProps, 'visible'> {
+    connections: {
+        name: string;
+        endpoint: string;
+    }[];
+}
+
+export const ConnectionModal: FC<ConnectionModalProps> = ({ title = 'Select connection', onCancel, ...props }) => {
+    const { visible, setVisible } = useConnectionModal();
+    const setConnection = useSetConnection();
+    const [expanded, setExpanded] = useState(false);
+
+    const handleCancel = useCallback(
+        (event: MouseEvent<HTMLElement>) => {
+            if (onCancel) onCancel(event);
+            if (!event.defaultPrevented) setVisible(false);
+        },
+        [onCancel, setVisible]
+    );
+
+    const handleConnectionClick = useCallback(
+        (
+            event: MouseEvent<HTMLElement>,
+            connection: {
+                name: string;
+                endpoint: string;
+            }
+        ) => {
+            // TODO: Add a way to replace commitment.
+            setConnection(new Connection(connection.endpoint, { commitment: 'confirmed' }));
```

Comment:
I'm not sure why this doesn't use the connection options provided by the existing ConnectionProvider context.